contracts/test/emergency/StarkNetValidator.test.ts

```diff
@@ -52,24 +75,281 @@ describe('StarkNetValidator', function () {
     mockStarkNetMessaging = await mockStarkNetMessagingFactory.deploy()
     await mockStarkNetMessaging.deployed()
 
+    // Deploy the mock feed
+    mockGasPriceFeed = await deployMockContract(deployer, aggregatorAbi)
+    await mockGasPriceFeed.mock.latestRoundData.returns(
+      '73786976294838220258' /** roundId */,
+      '96800000000' /** answer */,
+      '163826896' /** startedAt */,
+      '1638268960' /** updatedAt */,
+      '73786976294838220258' /** answeredInRound */,
+    )
+
+    // Deploy the mock access controller
+    mockAccessController = await deployMockContract(deployer, accessControllerAbi)
+
+    // Deploy the mock aggregator
+    mockAggregator = await deployMockContract(deployer, aggregatorAbi)
+    await mockAggregator.mock.latestRoundData.returns(
+      '73786976294838220258' /** roundId */,
+      1 /** answer */,
+      '163826896' /** startedAt */,
+      '1638268960' /** updatedAt */,
+      '73786976294838220258' /** answeredInRound */,
+    )
+  })
+
+  beforeEach(async () => {
     // Deploy the L1 StarkNetValidator
     const starknetValidatorFactory = await ethers.getContractFactory('StarkNetValidator', deployer)
-    starkNetValidator = await starknetValidatorFactory.deploy(
-      mockStarkNetMessaging.address,
-      l2Contract.address,
-    )
+    starkNetValidator = await starknetValidatorFactory
+      .connect(deployer)
```

Comment:
Deployed should be connected already as you used it in the constructor line before.